Source/WebCore/dom/MessageChannel.cpp

```diff
@@ -50,12 +50,12 @@ MessageChannel::MessageChannel(ScriptExecutionContext& context)
     : m_ports(generateMessagePorts(context))
 {
     if (!context.activeDOMObjectsAreStopped()) {
-        ASSERT(!port1().closed());
-        ASSERT(!port2().closed());
+        ASSERT(!port1().isDetached());
+        ASSERT(!port2().isDetached());
```

Comment:
I wonder if being in this state warrants any kind of logging to help us find bugs.